src/govuk/components/notification-banner/template.njk

```diff
@@ -0,0 +1,40 @@
+{% if params.role %}
+  {% set role = params.role %}
+{% elif params.type == "success" or params.type == "error" %}
+  {% set role = "alert" %}
+{% else %}
+  {% set role = "region" %}
+{% endif %}
+
+{% if params.tabindex %}
+  {% set tabindex = params.tabindex %}
+{% elif params.type == "success" or params.type == "error" %}
+  {% set tabindex = "-1" %}
+{% endif %}
+
+{% set titleHeadingLevel = params.titleHeadingLevel if params.titleHeadingLevel else 2 %}
+
+<div class="govuk-notification-banner
+  {%- if params.type == "success" %} govuk-notification-banner--success{% endif %}
+  {%- if params.type == "error" %} govuk-notification-banner--error{% endif %}
+  {%- if params.classes %} {{ params.classes }}{% endif %}"
+  role="{{ role }}"
+  {% if tabindex %} tabindex="{{ tabindex }}" {% endif %}
+  {% if not(params.type == "success") and not(params.type == "error") %} aria-labelledby="govuk-notification-banner-title" {% endif %}
+  {% if ((params.type == "success" or params.type == "error") and not(params.initialFocus === false)) or params.initialFocus === true %} data-module="govuk-initial-focus"{% endif %}
```

Comment:
This is quite hard to follow. Could we set it up at the top of the file? Maybe something like

```jinja
{% if 'initialFocus' in params %}
  {% set initialFocus = params.initialFocus %}
{% elif params.type == "success" or params.type == "error" %}
  {% set initialFocus = true %}
{% else %}
  {% set initialFocus = false %}
{% end %}
```

Reply:
I think the data-module should always be the component name, for example `data-module="govuk-notification-banner"`.

In case we want to add any other JS functionality in the future (like making banners dismissable) I'd suggest we always include the data-module attribute, and pass the initial focus as a separate data attribute – otherwise we're coupling the autofocusing behaviour with initialising the module.

```jinja
  data-module="govuk-initial-focus" {%- if initialFocus %} data-initial-focus="true"{% endif %}
```

Reply:
@36degrees We've usually name spaced the attribute value (`govuk-notification-banner`). If we're setting the value to a boolean I wonder if the attribute should be name spaced instead eg.
```
data-govuk-initial-focus="true"
```
 This feels like something we should have possibly considered when we decided on the name spacing syntax 🤔 

On another note, I'm thinking that this attribute should only be rendered if it's true to avoid a redundant 
```
data-govuk-initial-focus="false"
```
on the HTML of the default component. I think this could confuse teams unnecessarily and there's probably not any benefit on having it copied over. Or what do you think?

Reply:
I don't think we need to namespace the attributes. The `data-module` pattern is used elsewhere so it make sense to namespace that to avoid conflicts, but we only do anything with `data-initial-focus` when `data-module="govuk-notification-banner"` and at that point the element is already 'in our scope' and so we shouldn't have to worry about namespacing every option individually.

Assuming `data-initial-focus` should default to `false` (which it probably should do?), agree the macro can omit data-govuk-initial-focus entirely if it's false. I think that's what would happen in the example I gave in my previous comment?

Reply:
Okay that makes sense, thanks @36degrees. I think I've resolved this now, have a look.

Reply:
@hannalaakso Let me know if I'm interpreting these comments incorrectly, but I think the suggestion was not namespacing the initial-focus attribute because it should already be scoped inside the `data-module="govuk-notification-banner"` - but I think that attribute is missing?

The relevant bit from Ollie's comment that I'm looking at:
> we only do anything with data-initial-focus when data-module="govuk-notification-banner"